scipy/optimize/_linprog_ip.py

```diff
@@ -58,28 +69,35 @@ def _get_solver(sparse=False, lstsq=False, sym_pos=True, cholesky=True):
         Handle to the appropriate solver function
 
     """
-    if sparse:
-        if lstsq or not(sym_pos):
-            def solve(M, r, sym_pos=False):
-                return sps.linalg.lsqr(M, r)[0]
-        else:
-            # this is not currently used; it is replaced by splu solve
-            # TODO: expose use of this as an option
-            def solve(M, r):
-                return sps.linalg.spsolve(M, r, permc_spec="MMD_AT_PLUS_A")
+    try:
+        if sparse:
+            if lstsq or not(sym_pos):
+                def solve(r, sym_pos=False):
+                    return sps.linalg.lsqr(M, r)[0]
+            elif cholesky:
+                solve = cholmod(M)
+            else:
+                if has_umfpack:
+                    solve = sps.linalg.factorized(M)
+                else: # factorized doesn't pass permc_spec
+                    solve = sps.linalg.splu(M, permc_spec=permc_spec).solve
 
-    else:
-        if lstsq:  # sometimes necessary as solution is approached
-            def solve(M, r):
-                return sp.linalg.lstsq(M, r)[0]
-        elif cholesky:
-            solve = sp.linalg.cho_solve
         else:
-            # this seems to cache the matrix factorization, so solving
-            # with multiple right hand sides is much faster
-            def solve(M, r, sym_pos=sym_pos):
-                return sp.linalg.solve(M, r, sym_pos=sym_pos)
-
+            if lstsq:  # sometimes necessary as solution is approached
+                def solve(r):
+                    return sp.linalg.lstsq(M, r)[0]
+            elif cholesky:
+                L = sp.linalg.cho_factor(M)
+
+                def solve(r):
+                    return sp.linalg.cho_solve(L, r)
+            else:
+                # this seems to cache the matrix factorization, so solving
+                # with multiple right hand sides is much faster
+                def solve(r, sym_pos=sym_pos):
+                    return sp.linalg.solve(M, r, sym_pos=sym_pos)
+    except:
```

Comment:
This bare except is a little frightening. What is it intended to catch, and can we narrow its scope?

Reply:
@perimosocordiae On second thought, I don't think I can easily narrow this down. The point of all this is to try solving the linear system using successively robust solvers. Regardless of which solver fails and why (lots of possibilities) the next solver is attempted. Since there are many possibilities, I'm afraid that if we attempt to list all of them explicitly we risk missing some. In many cases this would cause the code to error out when it would have been fine if we had just caught any `Exception`.

Reply:
I caught `KeyboardInterrupt` and re-raised it.